### PR TITLE
Related location follow up

### PR DESCRIPTION
**Pull request by emord**
commit by commit

Fixes a couple of bugs and adds a test for the restore.

### CODE REVIEW

File: corehq/apps/locations/forms.py [hunk 1]
```diff
@@ -651,8 +653,11 @@ def __init__(self, domain, location, *args, **kwargs):
         self.helper.field_class = 'col-sm-9 col-md-8 col-lg-6'
 
     def save(self):
-        selected_location_ids = set(self.cleaned_data['related_locations'].split(','))
-        previous_locations = self.location.related_location_ids
+        selected_location_ids = set()
+        if self.cleaned_data['related_locations']:
+            # ''.split(',') == ['']
+            selected_location_ids = set(self.cleaned_data['related_locations'].split(','))
```
Comment (esoergel) [inline corehq/apps/locations/forms.py]: If the issue is that `self.cleaned_data['related_locations'] == ','`, then this wouldn't work, would it?  (`bool(',')` is `True`)

What about throwing a `filter(None, ...)` in there?

File: corehq/apps/locations/tests/test_location_fixtures.py [hunk 2]
```diff
@@ -647,6 +652,42 @@ def test_include_only_location_types(self):
...
 
+@flag_enabled("RELATED_LOCATIONS")
+@mock.patch.object(Domain, 'uses_locations', lambda: True)  # removes dependency on accounting
+class RelatedLocationFixturesTest(LocationHierarchyTestCase, FixtureHasLocationsMixin):
+    """
+    - State
+        - County
+            - City
+        - Region
+            - Town
```
Comment (esoergel) [inline corehq/apps/locations/tests/test_location_fixtures.py]: Did you intend to add Region and Town to the tests?

File: corehq/apps/locations/forms.py (hunk 1 above)
Comment (emord) [reply corehq/apps/locations/forms.py]: the issue was when the return value was an empty string `''`. ',' shouldn't be returned from select2 widget, but fixed it here: https://github.com/dimagi/commcare-hq/pull/21431/commits/2820c93ca52252dd4bd2476c9ba859db15e56a8d

File: corehq/apps/locations/fixtures.py [hunk 3]
```diff
@@ -229,8 +234,9 @@ def __call__(self, restore_state):
             return []
 
         restore_user = restore_state.restore_user
-        primary_location = restore_user.get_commtrack_location_id()
-        related_location_ids = SQLLocation.objects.get(location_id=primary_location).related_location_ids
+        user_locations = restore_user.get_sql_locations(restore_user.domain)
+        user_location_ids = user_locations.values_list('location_id', flat=True)
+        related_location_ids = LocationRelation.to_locations(user_location_ids)
```
[the remark this answers is not in the document]
Comment (emord) [reply corehq/apps/locations/fixtures.py]: Should be a little clearer here: https://github.com/dimagi/commcare-hq/pull/21431/commits/a7f4b254e1f4f04b93b534a9f0502fb580bd7caa

File: corehq/apps/locations/models.py [hunk 4]
```diff
@@ -828,6 +827,15 @@ class LocationRelation(models.Model):
     location_b = models.ForeignKey(
         SQLLocation, on_delete=models.CASCADE, related_name="+", to_field='location_id')
 
+    @classmethod
+    def to_locations(cls, location_ids):
```
[the remark this answers is not in the document]
Comment (emord) [reply corehq/apps/locations/models.py]: Should be a little clearer here: https://github.com/dimagi/commcare-hq/pull/21431/commits/a7f4b254e1f4f04b93b534a9f0502fb580bd7caa

File: corehq/apps/locations/tests/test_location_fixtures.py (hunk 2 above)
Comment (emord) [reply corehq/apps/locations/tests/test_location_fixtures.py]: https://github.com/dimagi/commcare-hq/pull/21431/commits/09faad71a20c3eb6b31a380da7d7580de3d221c7

**Review by esoergel — COMMENTED** [review]
Thanks for the followup - I think my only outstanding comment is https://github.com/dimagi/commcare-hq/pull/21431#discussion_r208011055

**Review by esoergel — APPROVED** [review]
offlined, looks good to me!

**Comment by emord** [comment]
looks like stickler had some random error that caused it to not start the job

**Pull request merged**